packages/apps/dashboard/server/src/modules/stats/stats.service.ts

```diff
@@ -34,13 +36,27 @@ export class StatsService implements OnModuleInit {
     private readonly envConfigService: EnvironmentConfigService,
     private readonly httpService: HttpService,
     private readonly storageService: StorageService,
-  ) {}
+    private schedulerRegistry: SchedulerRegistry,
+  ) {
+    if (this.envConfigService.hCaptchaStatsEnabled === true) {
+      const job = new CronJob('*/15 * * * *', () => {
+        this.fetchTodayHcaptchaStats();
+      });
+
+      this.schedulerRegistry.addCronJob('fetchTodayHcaptchaStats', job);
+      job.start();
+    }
+  }
 
   async onModuleInit() {
     const isHistoricalDataFetched = await this.isHistoricalDataFetched();
     const isHmtGeneralStatsFetched = await this.isHmtGeneralStatsFetched();
     const isHmtDailyStatsFetched = await this.isHmtDailyStatsFetched();
-    if (!isHistoricalDataFetched) {
+
+    if (
+      this.envConfigService.hCaptchaStatsEnabled === true &&
```

Comment:
@portuu3  that makes sense, thanks for details

Should we still fix the `aggregatedStats` calc?

Reply:
@dnechay what is the issue here?

Reply:
@portuu3 this code fails in case it reaches this line, because `reduce` method accepts `accumulator` as a second argument and it's undefined in this case, so `acc.solved += stat.solved;` is `undefined.solved += stat.solved;` and it fails